src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((accum: GroupsMap<T>, item) => {
+    const itemKey = item[key];
+    const result = accum;
```

Comment:
you can avoid creating `result` variable in this case, and do manipulation with `accum` (you don't reassign value entirely like this `accum = { someProperty: someValue }`)